packages/editor/src/components/post-actions/set-as-homepage.js

```diff
@@ -21,21 +21,37 @@ import { getItemTitle } from '../../utils/get-item-title';
 const SetAsHomepageModal = ( { items, closeModal, onActionPerformed } ) => {
 	const [ item ] = items;
 	const pageTitle = getItemTitle( item );
-	const { currentHomePage, showOnFront } = useSelect( ( select ) => {
+	const { showOnFront, pageOnFront } = useSelect( ( select ) => {
 		const { getEntityRecord } = select( coreStore );
 		const siteSettings = getEntityRecord( 'root', 'site', undefined );
-		const pageOnFront = siteSettings?.page_on_front;
 		return {
-			currentHomePage: getEntityRecord( 'postType', 'page', pageOnFront ),
 			showOnFront: siteSettings?.show_on_front,
+			pageOnFront: siteSettings?.page_on_front,
 		};
 	} );
+	const [ currentHomePageTitle, setCurrentHomePageTitle ] = useState( null );
```

Comment:
Not sure I understand entirely why this local state is needed? Also, wanted to share that we now have a `getHomePage` selector that returns `{ postType, postId }` of the current home page. If a static page is set postType will be `page` otherwise it will be `wp_template`

Reply:
Oh cool, I've added the `getHomePage` selector here: https://github.com/WordPress/gutenberg/pull/65426/commits/43dd85f0f4bafeb1ec7a275024787d14b3d86caa.

I think the local state needs to be used so that the value is updated based on when the `currentHomePage` value changes, otherwise the title value needs to be stored in a `useRef` which isn't updated when the `currentHomePage` value is updated. I'm not sure of a way to get around this without using local state.